### ISSUE TITLE
[core]Decompile error, missing a lot of code

### ISSUE THREAD

**Issue by bhwxha** [skylot/jadx#1671]
link:https://drive.google.com/file/d/1w3ZTTSDu54qrSfao7KfDlmffId3MKTWO/view?usp=sharing
class:`com.meituan.android.common.mtguard.wtscore.plugin.sign.core.a`
method:`private static URI a(Context context, URI uri, boolean z, byte[] bArr, String str, String str2, Map<String, String> map, String str3, Map<String, String> map2)`

jadx decompile error, by comparing, you can find a lot of missing code.

![image](https://user-images.githubusercontent.com/107301547/190857650-989bd803-8032-4997-9b39-a1756c1b873c.png)

![image](https://user-images.githubusercontent.com/107301547/190857684-1b5ac403-afc7-4a3c-bd31-80cec03c4b32.png)

**Comment by jpstotz** [comment]
Which Jadx version do you use?

**Comment by bhwxha** [comment]
> Which Jadx version do you use?

https://github.com/skylot/jadx/commit/b50706505f3f5ea562239b68710f9ab34e55a570

openjdk8

**Comment by jpstotz** [comment]
@bhwxha No, I was simply asking because you forgot to mention it. As you may have noticed there quite a few Jadx issue reports open. Unfortunately I don't have the experience to help skylot with the decompiler core system, thus I am trying to focus on checking issues are relevant and e.g. do not refer an outdated Jadx version. 

Regarding the code snippets you have posted I have to admit that I don't see some invocations both versions have in common.  So the only outcome I can derive from them is that there might be something wrong with the decompiled code of the mentioned method.

**Comment by bhwxha** [comment]
@skylot You can search for "System.arraycopy" in this class and you will find the following.
```
    /* JADX WARN: Code restructure failed: missing block: B:45:0x0171, code lost:
        r0 = r0.getBytes();
        r3 = java.lang.Math.min(r1.b.c().length, 16200);
        r4 = new byte[r0.length + r3];
        java.lang.System.arraycopy(r0, 0, r4, 0, r0.length);
        java.lang.System.arraycopy(r1.b.c(), 0, r4, r0.length, r3);
        r0 = r4;
     */
```

jeb

![image](https://user-images.githubusercontent.com/107301547/222629477-85bb641d-5372-49a6-b10c-549ccafbd16c.png)

This, shows that the pseudo-code is indeed missing quite a bit.
